subsys/bluetooth/host/att.c

```diff
@@ -2632,7 +2634,7 @@ static void att_reset(struct bt_att *att)
 		node = sys_slist_get_not_empty(&att->reqs);
 		req = CONTAINER_OF(node, struct bt_att_req, node);
 		if (req->func) {
-			req->func(NULL, BT_ATT_ERR_UNLIKELY, NULL, 0,
+			req->func(conn, BT_ATT_ERR_UNLIKELY, NULL, 0,
```

Comment:
Alternative solution:

Move the `att->conn = NULL` to after this loop, and use `att->conn` here instead. 

That seems simpler to me, and should provide the same result. 

Of course, there's not really any way for the application to know that this "error" is occurred due to a disconnect. I wonder if there's anything we can do there.

Reply:
I assumed there was a reason for why `att->conn = NULL` is above the callback invocation. Is it really the same?

Reply:
Hmm, it's a good question. I would expect that any connection related checks would check for the `conn->state` and not if `att->conn == NULL`, but it's hard to tell, and would require some additional time to check. 

There's no comment for this assignement, and @jhedberg can't remember (it is old code after all) the reasoning either :)

Reply:
> I would expect that any connection related checks would check for the `conn->state` and not if `att->conn == NULL`

If that's the case, `att->conn = NULL` is not needed at all, right? `att` is freed at the end of this function.

Reply:
Does `k_mem_slab_free(&att_slab, (void **)&att);` memset `att` to `0`? Not familiar with that function.

Reply:
I don't think it does. But `bt_att_accept`, memsets it on allocation.

Reply:
OK, then the question is if we ever check `att->conn`, and if it needs to be set to NULL here :)